### PR TITLE
ARROW-1216: [Python] Fix creating numpy array from arrow buffers on python 2

### PR DESCRIPTION
**Pull request by pcmoritz**

### CODE REVIEW

**Review by xhochy — APPROVED** [review]
+1, LGTM. I'll merge on green build.

Thanks for fixing this so quickly!

**Comment by pcmoritz** [comment]
No problem! It's a little unfortunate that numpy doesn't support the new buffer protocol (seems like in python 2.7 that would be possible in principle).

File: python/pyarrow/tests/test_io.py [hunk 1]
```diff
@@ -160,6 +160,15 @@ def test_buffer_bytearray():
     assert result == val
 
 
+def test_buffer_numpy():
+    # Make sure creating a numpy array from an arrow buffer works
+    byte_array = bytearray(20)
+    byte_array[0] = 42
+    buf = pa.frombuffer(byte_array)
+    array = np.frombuffer(buf, dtype="uint8")
+    assert array[0] == byte_array[0]
```
Comment (wesm) [inline python/pyarrow/tests/test_io.py]: Can you verify that this sets the array's base member?

Comment (pcmoritz) [inline python/pyarrow/tests/test_io.py]: @wesm Thanks for the note, I added the check and the tests have passed.

**Review by xhochy — APPROVED** [review]

**Pull request closed without merging**